### ISSUE TITLE
QUIC: thread 'tokio-runtime-worker' panicked at 'called Option::unwrap() .. quic

### ISSUE THREAD

**Issue by huglester** [solana-labs/solana#26263]
#### Problem
got a panic with solana testnet, 1.10.27 version

#### Proposed Solution

```
Jun 27 14:29:06 solana-validator[899043]: thread 'tokio-runtime-worker' panicked at 'called Option::unwrap() on a None value', streamer/src/nonblocking/quic.rs:420:65
Jun 27 14:29:06 solana-validator[899043]: stack backtrace:
Jun 27 14:29:06 solana-validator[899043]:    0: rust_begin_unwind
Jun 27 14:29:06 solana-validator[899043]:              at ./rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
Jun 27 14:29:06 solana-validator[899043]:    1: core::panicking::panic_fmt
Jun 27 14:29:06 solana-validator[899043]:              at ./rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
Jun 27 14:29:06 solana-validator[899043]:    2: core::panicking::panic
Jun 27 14:29:06 solana-validator[899043]:              at ./rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:48:5
Jun 27 14:29:06 solana-validator[899043]:    3: solana_streamer::nonblocking::quic::ConnectionTable::prune_oldest
Jun 27 14:29:06 solana-validator[899043]:    4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jun 27 14:29:06 solana-validator[899043]:    5: tokio::runtime::task::harness::poll_future
Jun 27 14:29:06 solana-validator[899043]:    6: tokio::runtime::task::harness::Harness<T,S>::poll
Jun 27 14:29:06 solana-validator[899043]:    7: std::thread::local::LocalKey<T>::with
Jun 27 14:29:06 solana-validator[899043]:    8: tokio::runtime::thread_pool::worker::Context::run_task
Jun 27 14:29:06 solana-validator[899043]:    9: tokio::runtime::thread_pool::worker::Context::run
Jun 27 14:29:06 solana-validator[899043]:   10: tokio::macros::scoped_tls::ScopedKey<T>::set
Jun 27 14:29:06 solana-validator[899043]:   11: tokio::runtime::thread_pool::worker::run
Jun 27 14:29:06 solana-validator[899043]:   12: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
Jun 27 14:29:06 solana-validator[899043]:   13: tokio::runtime::task::harness::Harness<T,S>::poll
Jun 27 14:29:06 solana-validator[899043]:   14: tokio::runtime::blocking::pool::Inner::run
Jun 27 14:29:06 solana-validator[899043]: note: Some details are omitted, run with RUST_BACKTRACE=full for a verbose backtrace.
Jun 27 14:29:06 solana-validator[899043]: [2022-06-27T14:29:06.290323637Z ERROR solana_metrics::metrics] datapoint: panic program="validator" thread="tokio-runtime-worker" one=1i message="panicked at 'called `O>
Jun 27 14:29:08 systemd[1]: solana.service: Main process exited, code=killed, status=11/SEGV
Jun 27 14:29:08 systemd[1]: solana.service: Failed with result 'signal'.
Jun 27 14:29:14 systemd[1]: solana.service: Scheduled restart job, restart counter is at 1.
Jun 27 14:29:14 systemd[1]: Stopped Solana TdS.

```

**Comment by lijunwangs** [comment]
This crash condition indicate the total_size is not reflecting the real connection counts. The following code is probably suspicious:

**Comment by pgarg66** [comment]
https://github.com/solana-labs/solana/pull/26073 has a fix for something related. It's not merged to v1.10 though. I'll cherry pick on v1.10.
